ttnn/cpp/ttnn/operations/experimental/cnn/convert_to_chw/device/convert_to_chw_program_factory.cpp

```diff
@@ -20,20 +20,29 @@ operation::ProgramWithCallbacks multi_core_convert_to_chw(
     const auto HW = input_shape[2];
     const auto C = input_shape[3];
 
+    tt::log_debug(tt::LogType::LogOp, "Running op with HW={}, C={}, shard_shape={}", HW, C, a.shard_spec()->shape);
+
     TT_ASSERT(C < TILE_HEIGHT, "C must be 32 or smaller");
 
-    const uint32_t c_tiles = 1;  // assume C <= 32
-    const uint32_t hw_tiles = HW / TILE_HEIGHT;
-    const uint32_t total_tiles = hw_tiles * c_tiles;
-    const uint32_t total_tiles_per_core = hw_tiles * c_tiles / input_cores.size();
+    const uint32_t total_tiles = HW / TILE_HEIGHT;  // assume C < 32
+    const uint32_t total_tiles_per_core = tt::div_up(total_tiles, input_cores.size());
```

Comment:
What happens if the grid stored in shard spec is not the actual grid that shards occupy? I think `input_cores` is prone to error here. I would add assert against actual expected number of shards based on input shape and shard shape.

Reply:
Added an assert checking expected number of shards vs. actual number